src/video_core/renderer_opengl/renderer_opengl.cpp

```diff
@@ -150,6 +151,14 @@ void RendererOpenGL::SwapBuffers() {
     {
         auto aggregator = Common::Profiling::GetTimingResultsAggregator();
         aggregator->AddFrame(profiler.GetPreviousFrameResults());
+        if (VideoCore::g_toggle_framelimit_enabled) {
+            Common::Profiling::Duration frame_time =
+                aggregator->GetAggregatedResults().frame_time.avg;
+            using LongMs = std::chrono::duration<long, std::chrono::microseconds::period>;
+            std::chrono::microseconds average_frame_time =
+                std::chrono::microseconds(std::chrono::duration_cast<LongMs>(frame_time).count());
+            FrameLimiter(average_frame_time, std::chrono::microseconds(1000000) / 60);
+        }
```

Comment:
it is a few ms slower than the previous implementation.lets take this, the expected frame time is 1s/60 , which is 16.666666666666... ms
now  taking the chrono units it will be
16ms , 16666uc, 1666666ns due to truncation . http://www.informit.com/articles/article.aspx?p=1881386&seqNum=2 . the nanosecond is more accurate. but even when there is no sleep, the is a delay between two frames, due to other instructions during their own thing. the earlier it wakes up, the earlier it executes those instructions, so it seems faster.
somehow , using floats or uint32 keeps the average frame time the closest to 1s/60 . reducing the amount of duration casts ( to 1 cast before function call and none in the function) does not give as good a result as the previous one. tested different situations here.